applications/CipherEX.md

```diff
@@ -0,0 +1,82 @@
+## Project Overview :page_facing_up:
+### Overview
+This project deliver a privacy preserving data sharing pallet(CipherEX) based on proxy re-encryption technology in Polkadot Ecosystem. 
+
+ If someone want to send a secret data to another, the tradition solution is using Diffie-Hellman Key exchange to generate a shared key and then they can encode the secret data using the shared key with symmetric encryption scheme such AES. 
+
+Instead of  Diffie-Hellman Key exchange, in our proposal we will use a threshold proxy re-encryption technology to transport the symmetric key which used for symmetric encryption. The idea comes from this [paper](https://github.com/nucypher/umbral-doc/blob/master/umbral-doc.pdf) .  And based on this ,we could design a "fair exchange" of secret data under a "Proxy-exist" setting.
+
+
+
+### Project Description
++ Suppose Alice want to sell a piece of secret data to Bob in exchange of Dot in Polkadot Ecosystem. Alice first encapsulate  his public key(pk_a) to generate a symmetric key(Skey) and a Capsule. After that she could use Skey to encode his secret into EncData using symmetric encryption scheme. And once this preparation is done she could upload EncData and Capsule in a storage layer, say IPFS.
++ Now if Bob want to purchase this secret data, he need to send a Lock transaction in parachain. This lock transaction would include Bob's public key(pk_b) and lock some token from Bob. This lock transaction could be unlocked once it accumulate enough unlock transactions from collator(also proxy in our setting) .
++ When Alice see Bob's lock transaction, she would generate re-encrypt key using Bob's Public key. Here we use Shamir's secret sharing to divide the key into N pieces(key_1,key_2,...key_N), and later Bob need m pieces to recover Skey. And Alice need to send these key to N randomly select collator(could use Follow-The-Satoshi algorithm to select collator here).
++ The collator need to  Re-Encapsulate key_i and capsule into cipher_i and send a unlock transaction in parachain. The unlock transaction contain the information of cipher_i and his signature. 
++ When there are enough unlock transaction in the parachain ,the locked token would be issue to Alice and at the same time Bob would be able to collect enough cipher_i(at least m).
++ Bob first download the EncData and Capsule from IPFS.  Then he could recover Skey using his private key (sk_b), Capsule and m pieces of cipher_i. Once Bob recover Skey, he could use it to decode EncData. Finally, Bob get the secret data from Alice. And Alice get token from Bob.
++ The description figure of the whole procedure is here:
+ ![dci](https://github.com/pawnz0/Documentation/blob/2cead2c261fc4c6930319e8684bc8b509608ab82/Figures/exchange.png?raw=true)
+
++ We plan to choose the sr25519 curve for our EC curve since it's a pairing-friendly curve and compatible with Polkadot system. And we would choose the AES-256-GCM scheme for the symmetric-key encryption scheme. 
```

Comment:
Ristretto is Ed25519 under the hood, no pairings.  We've no pairings on substrate currently, but maybe by the end of the summer.